cmd/goal/box.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"net/url"
+
+	"github.com/spf13/cobra"
+)
+
+var boxName string
+
+func init() {
+	appCmd.AddCommand(appBoxCmd)
+
+	appBoxCmd.AddCommand(appBoxInfoCmd)
+	appBoxCmd.AddCommand(appBoxListCmd)
+	appBoxCmd.PersistentFlags().Uint64Var(&appIdx, "app-id", 0, "Application ID")
+	appBoxCmd.MarkFlagRequired("app-id")
+
+	appBoxInfoCmd.Flags().StringVarP(&boxName, "name", "n", "", "Application box name. Use the same form as app-arg to name the box.")
+	appBoxInfoCmd.Flags().SetInterspersed(false)
+	appBoxInfoCmd.MarkFlagRequired("name")
+}
+
+var appBoxCmd = &cobra.Command{
+	Use:   "box",
+	Short: "Read application box data",
+	Args:  cobra.ArbitraryArgs,
+	Run: func(cmd *cobra.Command, args []string) {
+		// If no arguments passed, we should fallback to help
+		cmd.HelpFunc()(cmd, args)
+	},
+}
+
+var appBoxInfoCmd = &cobra.Command{
+	Use:   "info",
+	Short: "Retrieve information about an application box",
+	Args:  validateNoPosArgsFn,
+	Run: func(cmd *cobra.Command, args []string) {
+		_, client := getDataDirAndClient()
+
+		// Ensure box name is specified
+		if boxName == "" {
+			reportErrorf(errorMissingBoxName)
+		}
+
+		// Get box info
+		rawName, err := newAppCallBytes(boxName).raw()
+		if err != nil {
+			reportErrorf("Could not decode box name %s: %v", boxName, err)
+		}
+		urlEncoded := url.PathEscape(string(rawName))
```

Comment:
Do we still need the URL encoding in goal?

Reply:
I'd assume not based on the following struct tag:
https://github.com/jdtzmn/go-algorand/blob/13ad60f98a69473ea9b0524368e0882ab9456508/daemon/algod/api/client/restClient.go#L438